lib/Model.js

```diff
@@ -398,6 +406,40 @@ Model.get = function(NewModel, key, options, next) {
   return deferred.promise.nodeify(next);
 };
 
+Model.prototype.populate = function (options, resultObj, fillPath) {
+  if (!fillPath) {
+    fillPath = [];
+  }
+  var deferred = Q.defer();
+  var self = this;
+  var Models = this.$__.table.base.models;
+  if (!resultObj) {
+    resultObj = self;
+  }
+  if (!options.path || !options.model) {
+    throw new Error('Invalid parameters provided to the populate method');
+  }
+  try {
```

Comment:
Awesome!!!

That'll avoid manually prefixing such as `'test-Dog6'`

I'll do it on the weekend